modules/apps/transfer/client/cli/tx.go

```diff
@@ -121,7 +133,42 @@ using the {packet-timeout-timestamp} flag. If no timeout value is set then a def
 	cmd.Flags().Uint64(flagPacketTimeoutTimestamp, defaultRelativePacketTimeoutTimestamp, "Packet timeout timestamp in nanoseconds from now. Default is 10 minutes. The timeout is disabled when set to 0.")
 	cmd.Flags().Bool(flagAbsoluteTimeouts, false, "Timeout flags are used as absolute timeouts.")
 	cmd.Flags().String(flagMemo, "", "Memo to be sent along with the packet.")
+	cmd.Flags().String(flagForwarding, "", "Forwarding information in the form of a comma separated list of portID/channelID pairs, denoting the intermediary hops. If forwarding is specified any memo set will be included in the forwarding information created.")
+	cmd.Flags().Bool(flagUnwind, false, "Flag to indicate if the coin should be unwound to its native chain before forwarding.")
+
 	flags.AddTxFlagsToCmd(cmd)
 
 	return cmd
 }
+
+// parseForwarding parses the forwarding flag into a Forwarding object or nil if the flag is not specified. If the flag cannot
+// be parsed or the hops aren't in the portID/channelID format an error is returned.
+func parseForwarding(cmd *cobra.Command) (types.Forwarding, error) {
+	var hops []types.Hop
+
+	forwardingString, err := cmd.Flags().GetString(flagForwarding)
+	if err != nil {
+		return types.Forwarding{}, err
+	}
+	if strings.TrimSpace(forwardingString) == "" {
+		return types.Forwarding{}, nil
+	}
+
+	pairs := strings.Split(forwardingString, ",")
+	for _, pair := range pairs {
+		pairSplit := strings.Split(pair, "/")
```

Comment:
I guess if the port/channel pairs are messed up, then it would fail in the message validation. But maybe it's worth also running here the validation for port ID and channel ID.

Reply:
channel IDs shouldn't have slashes in them afaik

`var IsChannelIDFormat = regexp.MustCompile(`^channel-[0-9]{1,20}$`).MatchString`

Reply:
ah sorry I was mixing it up with denoms which can have slashes, we should be fine with this